BridgeAppSDK/SBASurveyFactory.swift

```diff
@@ -68,6 +68,12 @@ open class SBASurveyFactory : SBABaseSurveyFactory {
         return json["steps"] as? [NSDictionary]
     }()
     
+    // This is a time-consuming operation (b/c data groups have moved to the keychain)
+    // so use a lazy load to only load when called, but then retain the result in memory.
+    lazy open var currentDataGroups: Set<String> = {
```

Comment:
The most current data groups from Bridge can be directly accessed on the SBBStudyParticipant record through the "participantDataGroups" profile item, as a Set\<String\>. Access will be fast, as the study participant record is retrieved from CoreData and cached in memory when setting up Bridge initially, and then kept up-to-date with any changes.